### PR TITLE
cardano-node: reduce orphan instances

### PR DESCRIPTION
**Pull request by smelc**
Contributes to fixing https://github.com/input-output-hk/cardano-node/issues/5470

# Description

This PR starts reducing the content of the `Cardano.Node.Orphans` module.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated.  These may include:
  - golden tests
  - property tests
  - roundtrip tests
  - integration tests
  See [Runnings tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Any changes are noted in the `CHANGELOG.md` for affected package
- [ ] The version bounds in `.cabal` files are updated
- [ ] CI passes. See note on CI.  The following CI checks are required:
  - [ ] Code is linted with `hlint`.  See `.github/workflows/check-hlint.yml` to get the `hlint` version
  - [ ] Code is formatted with `stylish-haskell`.  See `.github/workflows/stylish-haskell.yml` to get the `stylish-haskell` version
  - [ ] Code builds on Linux, MacOS and Windows for `ghc-8.10.7` and `ghc-9.2.7`
- [ ] Self-reviewed the diff

### CODE REVIEW

File: cardano-node/src/Cardano/Node/Tracing/Tracers/Peer.hs [hunk 1]
```diff
@@ -81,7 +82,7 @@ ppInFlight f = printf
  (ppMaxSlotNo $ peerFetchMaxSlotNo f)
  (peerFetchReqsInFlight f)
  (Set.size $ peerFetchBlocksInFlight f)
- (peerFetchBytesInFlight f)
+ (getSizeInBytes $ peerFetchBytesInFlight f)
```
Comment (Jimbo4350) [inline cardano-node/src/Cardano/Node/Tracing/Tracers/Peer.hs:85]: Why this change?

**Review by Jimbo4350 — COMMENTED** [review]
LGTM, have you opened PRs in the other relevant repos?

File: cardano-node/src/Cardano/Node/Tracing/Tracers/Peer.hs (hunk 1 above)
Comment (smelc) [reply cardano-node/src/Cardano/Node/Tracing/Tracers/Peer.hs:85]: This location too was relying on the orphan instance that is being deleted, and I now need to unpack the data.

**Comment by smelc** [comment]
> LGTM, have you opened PRs in the other relevant repos?

@Jimbo4350> That's my question for the remaining instances. For those I can't inline the typeclass definition in the positions where it is used. So should I add the non-orphans instances to the other relevant repos? And then we will use the new instances when we update our dependencies (that's our workflow right?).

**Comment by smelc** [comment]
@Jimbo4350> I started doing the PRs to other repos, e.g. https://github.com/input-output-hk/iohk-monitoring-framework/pull/644

However, I suggest to merge this one soon, as this one - in the current state - is standalone. What do you think?

**Comment by smelc** [comment]
Created the other PR required to get rid of `Orphans.hs`: https://github.com/input-output-hk/ouroboros-network/pull/4674

File: cardano-node/src/Cardano/Node/Tracing/Tracers/Peer.hs (hunk 1 above)
Comment (carbolymer) [reply cardano-node/src/Cardano/Node/Tracing/Tracers/Peer.hs:85]: I think the `PrintfArg` instance should be just moved to `ouroboros-network` if possible, and this place should stay as it was.

Comment (Jimbo4350) [reply cardano-node/src/Cardano/Node/Tracing/Tracers/Peer.hs:85]: I agree lets leave this as it is. I don't want to accidentally change the tracing format

**Review by Jimbo4350 — APPROVED** [review]

File: cardano-node/src/Cardano/Node/Tracing/Tracers/Peer.hs (hunk 1 above)
Comment (smelc) [reply cardano-node/src/Cardano/Node/Tracing/Tracers/Peer.hs]: I reverted the change :+1:

**Comment by smelc** [comment]
I'm enqueueing for merge, since in the final state it only removes an unusued orphan instance and is good to go as is.

I've kept track in https://github.com/input-output-hk/cardano-node/issues/5470 of the follow-up work in other repos to further shrink the instances from this repo.

**Review by carbolymer — APPROVED** [review]

**Comment by carbolymer** [comment]
@smelc I have no clue what happened to ubuntu runners, but maybe you need to add `apt update` in GHA workflow to fix them: https://github.com/input-output-hk/cardano-api/pull/248/files#diff-c09cc8dc973415b1221308559b96942a1c4b872e690bf360fc22d9b42822d519R48
